src/cpp/fastdds/publisher/DataWriterImpl.cpp

```diff
@@ -1144,6 +1135,29 @@ bool DataWriterImpl::can_qos_be_updated(
     return updatable;
 }
 
+void DataWriterImpl::set_fragment_size_on_change(
+        WriteParams& wparams,
+        CacheChange_t* ch,
+        const uint32_t& high_mark_for_frag)
+{
+    uint32_t final_high_mark_for_frag = high_mark_for_frag;
+
+    // If needed inlineqos for related_sample_identity, then remove the inlinqos size from final fragment size.
+    if (wparams.related_sample_identity() != SampleIdentity::unknown())
+    {
+        final_high_mark_for_frag -= 32;
```

Comment:
I'd add this `32` as a `constexpr` in DataMsg.hpp. I think right now it's only used here (and probably in PublisherImpl), but it will improve readability for sure. It'd look less like a magic number